components/Collections/src/MDCCollectionViewController.m

```diff
@@ -98,17 +91,31 @@ - (void)viewDidLoad {
   // Set up ink touch controller.
   _inkTouchController = [[MDCInkTouchController alloc] initWithView:self.collectionView];
   _inkTouchController.delegate = self;
+}
+
+- (void)viewWillLayoutSubviews {
+  [super viewWillLayoutSubviews];
+
+  // Add header/footer infoBars if editing is allowed.
+  if ([self collectionViewAllowsEditing:self.collectionView]) {
```

Comment:
Is switching between allowing editing and not allowing editing supported? If so, how are subclasses to handle that transition correctly? If not, why not?